File: Exemplos SmartPOS/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_SmartPOS_Flutter-font/lib/IntentDigitalHubService/ELGINPAY/Commands/inicia_venda_credito.dart
import 'package:flutter_smartpos/IntentDigitalHubService/ELGINPAY/elginpay_command.dart';

class IniciaVendaCredito extends ElginPayCommand {
  final String _valorTotal;
  final int _tipoFinanciamento;
  final int _numeroParcelas;

  IniciaVendaCredito(
      this._valorTotal, this._tipoFinanciamento, this._numeroParcelas)
      : super('iniciaVendaCredito');

  @override
  get functionParametersJson => {
        'valorTotal': _valorTotal,
        'tipoFinanciamento': _tipoFinanciamento,
        'numeroParcelas': _numeroParcelas,
      };
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_SmartPOS_ReactNative-font/src/intentDigitalHubService/ELGINPAY/Commands/IniciaVendaCredito.ts
import {ElginPayCommand} from '../ElginPayCommand';

export class IniciaVendaCredito extends ElginPayCommand {
  readonly valorTotal: string;
  readonly tipoFinanciamento: number;
  readonly numeroParcelas: number;

  constructor(
    valorTotal: string,
    tipoFinanciamento: number,
    numeroParcelas: number,
  ) {
    super('iniciaVendaCredito');
    this.valorTotal = valorTotal;
    this.tipoFinanciamento = tipoFinanciamento;
    this.numeroParcelas = numeroParcelas;
  }

  functionParameters(): object {
    return {
      valorTotal: this.valorTotal,
      tipoFinanciamento: this.tipoFinanciamento,
      numeroParcelas: this.numeroParcelas,
    };
  }
}
